src/cljc/matcher_combinators/core.cljc

```diff
@@ -503,6 +513,49 @@
        ::result/weight 1}))
   (-base-name [_] 'predicate))
 
+(defrecord AnyOf [matchers]
+  Matcher
+  (-matcher-for [this] this)
+  (-matcher-for [this _] this)
+  (-match [this actual]
+    (reduce (fn [min-mismatch-weight matcher]
+              (if (= ::end matcher)
+                {::result/type   :mismatch
+                 ::result/value  (model/->Mismatch (concat ['any-of] matchers) actual)
+                 ::result/weight min-mismatch-weight}
+                (let [result (match matcher actual)]
+                  (if (indicates-match? result)
+                    (reduced {::result/type   :match
+                              ::result/value  actual
+                              ::result/weight 0})
+                    (min (::result/weight result)
+                         min-mismatch-weight)))))
+            #?(:clj Integer/MAX_VALUE
+               :cljs (.-MAX_SAFE_INTEGER js/Number))
+            (conj (into [] matchers) ::end)))
+  (-base-name [_] 'any-of))
+
+(defrecord AllOf [matchers]
+  Matcher
+  (-matcher-for [this] this)
+  (-matcher-for [this _] this)
+  (-match [this actual]
+    (reduce (fn [_acc matcher]
+              (if (= ::end matcher)
+                {::result/type   :match
+                 ::result/value  actual
+                 ::result/weight 0}
+                (let [result (match matcher actual)]
+                  (when-not (indicates-match? result)
+                    (reduced
+                      {::result/type   :mismatch
+                       ::result/value  (model/->Mismatch (concat ['all-of] matchers) actual)
```

Comment:
note that mismatch info doesn't tell you want matched and what didn't. I imagine we could improve the info returned, but maybe would be nice to in a follow-up(?)